tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/passes/WatchdogPassImpl.scala

```diff
@@ -0,0 +1,46 @@
+package at.forsyte.apalache.tla.pp.passes
+
+import at.forsyte.apalache.infra.passes.{Pass, PassOptions, TlaModuleMixin}
+import at.forsyte.apalache.tla.lir.transformations.{LanguagePred, LanguageWatchdog}
+import com.google.inject.Inject
+import com.google.inject.name.Named
+import com.typesafe.scalalogging.LazyLogging
+
+class WatchdogPassImpl @Inject() (val options: PassOptions, val pred: LanguagePred,
```

Comment:
Perhaps it would be safer to run the watchdog before/after each transformation, since a mistake made in a transformation could make the module invalid to the language predicate. Even if there is only one pass for now, it seems better to have this incorporated in the existing pass than isolated on its own pass. WDYT?

Reply:
We could, but I think doing it before any other pass and then again in the last pass, i.e.
https://github.com/informalsystems/apalache/blob/274aeaaa557c0fdde11d3ef24f1d0a692735d355/tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/passes/ConstraintGenPassImpl.scala#L28-L38
suffices to catch out-of-fragment IR. The tradeoff is that the post-check won't tell you exactly which transformation broke out of the fragment, but figuring that out is easy once you start debugging.